dbms/src/Flash/Coprocessor/ChunkDecodeAndSquash.cpp

```diff
@@ -25,6 +26,58 @@ CHBlockChunkDecodeAndSquash::CHBlockChunkDecodeAndSquash(
 {
 }
 
+std::optional<Block> CHBlockChunkDecodeAndSquash::decodeAndSquashV1(std::string_view sv)
+{
+    if unlikely (sv.empty())
+    {
+        std::optional<Block> res;
+        if (accumulated_block)
+            res.swap(accumulated_block);
+        return res;
+    }
+
+    // read first byte of compression method flag which defined in `CompressionMethodByte`
+    if (static_cast<CompressionMethodByte>(sv[0]) == CompressionMethodByte::NONE)
+    {
+        ReadBufferFromString istr(sv.substr(1, sv.size() - 1));
+        return decodeAndSquashV1Impl(istr);
+    }
+
+    ReadBufferFromString istr(sv);
+    auto && compress_buffer = CompressedCHBlockChunkReadBuffer(istr);
+    return decodeAndSquashV1Impl(compress_buffer);
+}
+
+std::optional<Block> CHBlockChunkDecodeAndSquash::decodeAndSquashV1Impl(ReadBuffer & istr)
+{
+    std::optional<Block> res;
+
+    if (!accumulated_block)
+    {
+        size_t rows{};
+        Block block = DecodeHeader(istr, codec.header, rows);
+        if (rows)
```

Comment:
There is a potential problems here about the output's block row size:
consider that rows >> rows_limit, for example, rows = 10 * rows_limit, the decode will still return one block, which makes `rows_limit` meaningless.

Reply:
https://github.com/pingcap/tiflash/blob/eaf593f59348684d0e928692c755b0f03a6ac788/dbms/src/Flash/Coprocessor/ChunkDecodeAndSquash.cpp#L67-L73

The `rows_limit` is the lower bound to accumulate blocks.

https://github.com/pingcap/tiflash/blob/6060432a90f5915b2ce5d0e79602acb908969a1c/dbms/src/Flash/Mpp/HashPartitionWriter.cpp#L104-L106

In HashPartitionWriter, I have made the upper bound of batch to `8k * part-num rows` or `128MB mem size`,